go-update/src/main/java/org/reactome/release/goupdate/GoTermsUpdater.java

```diff
@@ -319,236 +255,272 @@ else if (goTermsFromFile.get(goID).containsKey(GoUpdateConstants.IS_OBSOLETE) &&
 		mainOutput.append(mismatchCount + " existing GO term instances in the database had mismatched categories when compared to the file (and were deleted from the database).\n");
 		mainOutput.append(obsoleteCount + " were obsolete. "+deletedCount+ " were actually deleted, and "+undeleteble.size()+" could not be deleted due to existing referrers.\n");
 		mainOutput.append(pendingObsoleteCount + " are pending obsolescence (and will probably be deleted at a future date).\n");
-		
-		reconcile(goTermsFromFile, goToECNumbers);
+		GoTermsReconciler reconciler = new GoTermsReconciler(this.adaptor);
+		reconciler.reconcile(goTermsFromFile, goToECNumbers);
 		
 		return mainOutput;
 	}
 
-	protected static Map<GKSchemaAttribute, Integer> getReferrerCounts(GKInstance inst) throws Exception
+	/**
+	 * Updates the relationships of GO terms.
+	 * @param goTermsFromFile - the GO terms from the GO file.
+	 * @param allGoInstances - a map of ALL GO instances from the database.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws InvalidAttributeValueException
+	 */
+	private void updateRelationships(Map<String, Map<String, Object>> goTermsFromFile, Map<String, List<GKInstance>> allGoInstances) throws InvalidAttributeException, Exception, InvalidAttributeValueException
```

Comment:
Is there any reason there are multiple Exception types? Since they all fall under Exception

Reply:
I don't like to hide more specific exceptions behind `Exception`. All three are listed because all three could be thrown by different parts of this function. I could simplify this by only throwing `Exception`, but then that could be misleading to future users of this function.